### PR TITLE
Added upgraded instructions and other stuff

### PR DESCRIPTION
**Pull request by lzap**
Proxy is no longer the recommended option. And 1.7 is the only compatible
version.

### CODE REVIEW

File: plugins/foreman_discovery/2.0/index.md [hunk 1]
```diff
@@ -23,11 +23,25 @@ direct access both to the provisioning network and Foreman instance:
     --
 
 Foreman Discovery does support "legacy" mode where communication is direct
-with no Foreman Proxy involved. See remarks in this documentation about how to
-configure this mode which is not recommended.
+with no Foreman Proxy involved. Currently (Foreman Discovery 2.0) this is
+still the recommended way to configure it due to technical limitations of
+Foreman Proxy 1.7 (this is explained bellow).
```
Comment (mmoll) [inline plugins/foreman_discovery/2.0/index.md]: below

Comment (lzap) [inline plugins/foreman_discovery/2.0/index.md]: Thanks found two more :-D

File: plugins/foreman_discovery/2.0/index.md [hunk 2]
```diff
@@ -160,6 +175,41 @@ To build a discovery image, please visit the foreman-discovery-image git
...
+
+### 2.4.1 Upgrade with Proxy Discovery plugin
+
+This method is *not recommended* for Discovery 2.0 because due to limitation
+of Foreman Proxy 1.7 it is not possible to configure this via HTTPS protocol.
+There are two possible workarounds. First, it is possible to run a separate
+http only proxy with only discovery plugin enabled dedicated to discovery
+communication. Second, it is possible to configure Proxy 1.7 with HTTPS using
+an [experimental (non-production)
+patch](https://github.com/theforeman/smart_proxy_discovery/pull/4).
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: Are you sure you want to link this?

File: plugins/foreman_discovery/2.0/index.md [hunk 3]
```diff
@@ -160,6 +175,41 @@ To build a discovery image, please visit the foreman-discovery-image git
...
+
+Follow Manual installation > Foreman Proxy Discovery plugin from above and
+proceed with the next upgrade step (Reboot discovered hosts).
+
+### 2.4.2 Upgrade without Proxy Discovery plugin (recommended)
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: Put the recommended steps _first_.

File: plugins/foreman_discovery/2.0/index.md [hunk 4]
```diff
@@ -273,7 +337,86 @@ chosen OS, and finally into the installed OS.
 
 Delete a machine and reboot it to have it move back to the Discovery Pool.
 
-# 4.1 Automatic provisioning
+## 4.3 Automatic provisioning
+
+Starting with version 2.0, it is possible to predefine provisioning rules
+which will assign hostgroup to provisioned hosts and trigger provisioning
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: nitpick "host group"

File: plugins/foreman_discovery/2.0/index.md [hunk 5]
```diff
@@ -273,7 +337,86 @@ chosen OS, and finally into the installed OS.
...
+* **Host Group** is assigned to a matching host before starting provisioning
+  process. It is very important the selected Host Group has all the required
+  parameters set (domain, subnet, root password), otherwise provisioning
+  process will fail.
+* **Hostname** defines a pattern to assign human-readable hostnames to the
+  matching hosts. When left empty, hostname is assigned in a form of
+  `macMACADDRESS` by default. The same syntax as for provisioning templates is
+  used. See below for more information and examples.
+* **Hosts limit** enables to limit maximum amount of provisioned hosts per
+  rule. If a limit was reached, rule is not taking effect until one or more
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: "reached, the rule will not take effect until"

File: plugins/foreman_discovery/2.0/index.md [hunk 6]
```diff
@@ -273,7 +337,86 @@ chosen OS, and finally into the installed OS.
...
+  hosts are deleted. Typical use case are rules per server rack or row when it
+  is necessary to change provisioning parameters like hostname or hostgroup
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: host group

File: plugins/foreman_discovery/2.0/index.md [hunk 7]
```diff
@@ -273,7 +337,86 @@ chosen OS, and finally into the installed OS.
...
+* **Enabled** flag is used for temporary shutdown of rules.
+
+Once some rules are defined, the good practice is to discover a host and apply
+the rules using Auto discover button on the host.
+
+By default, Foreman does not trigger auto discovery automatically. This must
+be explicitly turned on in
+*Administer > Settings > Discovered > discovery_auto*.
+
+## 4.3.1 Search syntax
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md:407]: You may also want to link to section 4.1.5 of the main manual

Comment (lzap) [inline plugins/foreman_discovery/2.0/index.md:407]: > You may also want to link to section 4.1.5 of the main manual

ALL done.

## 

Later,
 Lukas #lzap Zapletal

File: plugins/foreman_discovery/2.0/index.md [hunk 8]
```diff
@@ -160,6 +175,38 @@ To build a discovery image, please visit the foreman-discovery-image git
...
+
+### 2.4.1 Upgrade without Proxy Discovery plugin (recommended)
+
+Proceed with the next step (Reboot discovered hosts) and in the next section
+(Configuration) make sure the kernel line in the PXELinux default template has
+`proxy.type` set to `foreman` and `proxy.url` points to a *Foreman* instance.
+
+    APPEND ... proxy.url=https://YOUR_FOREMAN proxy.type=foreman
+
+### 2.4.1 Upgrade with Proxy Discovery plugin
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: 2.4.2

File: plugins/foreman_discovery/2.0/index.md [hunk 9]
```diff
@@ -23,11 +23,25 @@ direct access both to the provisioning network and Foreman instance:
...
+with no Foreman Proxy involved. Currently (Foreman Discovery 2.0) this is
+still the recommended way to configure it due to technical limitations of
+Foreman Proxy 1.7 (this is explained below).
 
 ## 1.1 Release Notes
 
+For each Foreman version there is particular version of Foreman Discovery
+plugin:
+
+| Foreman  | Plugin  | Image
```
Comment (domcleal) [inline plugins/foreman_discovery/2.0/index.md]: Formatting on this table is very plain... maybe check what we do in the main manual to make it clearer?

**Comment by GregSutcliffe** [comment]
+1 when all of Dominic's comments are addressed

**Comment by lzap** [comment]
> +1 when all of Dominic's comments are addressed

All done.

## 

Later,
 Lukas #lzap Zapletal

**Comment by domcleal** [comment]
Thanks @lzap!

**Pull request merged**